docs/proposal/20220712-garbage-collection.md

```diff
@@ -0,0 +1,339 @@
+last-updated: 2022-07-20
+status: implemented
+see-also:
+- https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1718
+- https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3518
+replaces: []
+superseded-by: []
+---
+
+# External Resource Garbage Collection
+
+## Table of Contents
+
+- [External Resource Garbage Collection](#external-resource-garbage-collection)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+      - [Story 3](#story-3)
+      - [Story 4](#story-4)
+  - [Requirements](#requirements)
+    - [Functional](#functional)
+    - [Non-Functional](#non-functional)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      - [Proposed Changes](#proposed-changes)
+        - [Cluster Creation](#cluster-creation)
+          - [API Changes](#api-changes)
+          - [Controller Changes](#controller-changes)
+          - [New Garbage Collection Service](#new-garbage-collection-service)
+        - [Cluster Deletion](#cluster-deletion)
+          - [Controller Changes](#controller-changes-1)
+          - [New Garbage Collection Service](#new-garbage-collection-service-1)
+        - [clusterawsadm changes](#clusterawsadm-changes)
+    - [Alternative Approaches Considered](#alternative-approaches-considered)
+      - [Using CCM to do the delete](#using-ccm-to-do-the-delete)
+    - [Risks and Mitigations](#risks-and-mitigations)
+      - [Replicating CCM](#replicating-ccm)
+      - [Similar functionality in upstream CAPI](#similar-functionality-in-upstream-capi)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+      - [Alpha](#alpha)
+      - [Beta](#beta)
+      - [Stable](#stable)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+- CAPA - An abbreviation of Cluster API Provider AWS.
+- ELB - Elastic Load Balancer
+- NLB - Network Load Balancer
+- CCM - Cloud Controller Manager
+
+## Summary
+
+If you create a child cluster using CAPA which then in turn creates a `Service` of type `LoadBalancer` this results in a load balancer being created in AWS for that service. The type of load balancer created by default is a **Classic ELB** but you can also create a NLB by annotating your service. For example:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: podinfo-nlb
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-type: "nlb"
+```
+
+If you try to delete the child/tenant cluster using CAPI/CAPA then it will fail to delete the clusters infrastructure fully in AWS as the VPC is still being using by the NLB that was created. For example:
```

Comment:
```suggestion
If you try to delete the child/tenant cluster using CAPI/CAPA then it will fail to delete the clusters infrastructure fully in AWS as the VPC is still being used by the NLB that was created. For example:
```

Reply:
Changed